cpp/include/raft/stats/trustworthiness_score.cuh

```diff
@@ -48,6 +50,37 @@ double trustworthiness_score(const raft::handle_t& h,
   return detail::trustworthiness_score<math_t, distance_type>(
     h, X, X_embedded, n, m, d, n_neighbors, batchSize);
 }
+
+/**
+ * @brief Compute the trustworthiness score
+ * @tparam value_t the data type
+ * @tparam idx_t Integer type used to for addressing
+ * @param[in] handle the raft handle
+ * @param[in] X: Data in original dimension
+ * @param[in] X_embedded: Data in target dimension (embedding)
+ * @param[in] n_neighbors Number of neighbors considered by trustworthiness score
+ * @param[in] batch_size Batch size
+ * @return[out] Trustworthiness score
+ */
+template <raft::distance::DistanceType distance_type, typename value_t, typename idx_t>
+double trustworthiness_score(const raft::handle_t& handle,
+                             raft::device_matrix_view<const value_t, idx_t, raft::row_major> X,
+                             raft::device_matrix_view<value_t, idx_t, raft::row_major> X_embedded,
```

Comment:
`X_embedded` is an input as well, which I don't think should be modified in this function. We shouldmark that `const value_t` as well.

Reply:
`X_embedded` is used as input to run a kNN, which is not taking const inputs as well, so it would require some changes on the `raft::spatial::knn`

Reply:
For now, you can accept the input as `const` and then do a `const_cast` inside the public API function (with a TODO and a note). The goal here is to make sure these arguments are sending the right message to the user, even if the data ends up being modified slightly in place (and then reverted back).